File: AppForms/AppForms.Web/Modules/ServerTypes/AppForms/FacultiesRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface FacultiesRow {
    Id?: number;
    Name?: string;
}

export abstract class FacultiesRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'AppForms.Faculties';
    static readonly deletePermission = 'Faculties';
    static readonly insertPermission = 'Faculties';
    static readonly readPermission = 'Faculties';
    static readonly updatePermission = 'Faculties';

    static readonly Fields = fieldsProxy<FacultiesRow>();
}

File: AppForms/AppForms.Web/Modules/AppForms/Faculties/FacultiesRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace AppForms.AppForms;

[ConnectionKey("AppForms"), Module("AppForms"), TableName("Faculties")]
[DisplayName("Faculties"), InstanceName("Faculties")]
[ReadPermission("Faculties")]
[ModifyPermission("Faculties")]
public sealed class FacultiesRow : Row<FacultiesRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id
    {
        get => fields.Id[this];
        set => fields.Id[this] = value;
    }

    [DisplayName("Name"), NotNull, QuickSearch, NameProperty]
    public string Name
    {
        get => fields.Name[this];
        set => fields.Name[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField Name;

    }
}